Improve the Basis class documentation

Calinou opened the pull request.

Calinou committed:
Improve the Basis class documentation

aaronfranke commented on doc/classes/Basis.xml:
```diff
@@ -78,6 +78,7 @@
 			<return type="Quat">
 			</return>
 			<description>
+				Assuming that the matrix is a proper rotation matrix (orthonormal matrix with determinant +1), return a quaternion. See [method get_euler] if you need Euler angles instead.
```
It seems that this method is self-orthonormalizing, so no assumptions are made. https://github.com/godotengine/godot/blob/master/core/math/basis.cpp#L370 (also I noticed the comments of these methods mention Euler angles when they shouldn't, but this can be fixed later if you don't want to)

Also, is it a good idea to point users towards `get_euler`? I think if users are specifically looking for Euler angles they will Ctrl+F for it, but in general Euler angles should be discouraged in my opinion.

Calinou commented:
I updated the pull request. I left a reference to `get_euler` in the `get_rotation_quat` method for completeness' sake, but I added a short disclaimer about Euler angles.

aaronfranke commented on doc/classes/Basis.xml:
```diff
@@ -64,7 +64,7 @@
 			</return>
 			<description>
-				Assuming that the matrix is a proper rotation matrix (orthonormal matrix with determinant +1), return Euler angles (in the YXZ convention: first Z, then X, and Y last). Returned vector contains the rotation angles in the format (X angle, Y angle, Z angle).
+				Returns the basis' Euler angles (in the YXZ convention: first Z, then X, and Y last). Returned vector contains the rotation angles in the format (X angle, Y angle, Z angle). See [method get_rotation_quat] if you need a quaternion instead.
```
This implies that the basis just "has" Euler angles, but in reality it has a rotation matrix which is then converted. I think it would make more sense as "Return the rotation of this basis in the form of Euler angles". Same thing below.

akien-mga replied:
@Calinou ^

aaronfranke commented on doc/classes/Basis.xml:
```diff
@@ -64,20 +64,21 @@
 			</return>
 			<description>
-				Assuming that the matrix is a proper rotation matrix (orthonormal matrix with determinant +1), return Euler angles (in the YXZ convention: first Z, then X, and Y last). Returned vector contains the rotation angles in the format (X angle, Y angle, Z angle).
+				Returns the basis' rotation in the form of Euler angles (in the YXZ convention: first Z, then X, and Y last). The returned vector contains the rotation angles in the format (X angle, Y angle, Z angle). See [method get_rotation_quat] if you need a quaternion instead.
```
I think `basis'` should be `basis's` since the ending apostrople is used for plurals with ownership.

akien-mga replied:
I didn't know about that rule, but it seems correct: https://www.thepunctuationguide.com/apostrophe.html#possessives

(And lol at English -- an exception for words only when followed by "sake"? Come on now.)

akien-mga commented:
Thanks!
